src/main/java/ca/corefacility/bioinformatics/irida/security/permissions/metadata/ReadProjectMetadataResponsePermission.java

```diff
@@ -0,0 +1,121 @@
+			MetadataRestrictionRepository metadataRestrictionRepository) {
+		this.userRepository = userRepository;
+		this.projectUserJoinRepository = projectUserJoinRepository;
+		this.metadataRestrictionRepository = metadataRestrictionRepository;
+	}
+
+	@Override
+	public String getPermissionProvided() {
+		return PERMISSION_PROVIDED;
+	}
+
+	@Override
+	public boolean isAllowed(Authentication authentication, Object targetDomainObject) {
+		//ensure the object we're checking is a ProjectMetadataResponse
+		if (!(targetDomainObject instanceof ProjectMetadataResponse)) {
+			throw new IllegalArgumentException(
+					"Object is not a ProjectMetadataResponse: " + targetDomainObject.getClass());
+		}
+		ProjectMetadataResponse metadataResponse = (ProjectMetadataResponse) targetDomainObject;
+
+		//get the user & projec
+		User user = userRepository.loadUserByUsername(authentication.getName());
+		Project project = metadataResponse.getProject();
+		ProjectUserJoin projectJoinForUser = projectUserJoinRepository.getProjectJoinForUser(project, user);
+
+		//get the user's role on the project
+		ProjectRole userProjectRole;
+		if (projectJoinForUser != null) {
+			userProjectRole = projectJoinForUser.getProjectRole();
+		} else if (user.getSystemRole()
+				.equals(Role.ROLE_ADMIN)) {
+			//if the user isn't on the project but is an admin, treat them as a project owner
+			userProjectRole = ProjectRole.PROJECT_OWNER;
+		} else {
+			//if the user is not otherwise on the project, they shouldn't be able to read anything
+			return false;
+		}
+
+		//get the metadata restrictions on the project
+		List<MetadataRestriction> restrictionForProject = metadataRestrictionRepository.getRestrictionForProject(
+				project);
+		Map<MetadataTemplateField, MetadataRestriction> restrictionMap = restrictionForProject.stream()
+				.collect(Collectors.toMap(MetadataRestriction::getField, field -> field));
+
+		//go through the metadata being returned and get a distinct collection of the fields
+		Map<Long, Set<MetadataEntry>> metadata = metadataResponse.getMetadata();
+		Set<MetadataTemplateField> fields = new HashSet<>();
+		for (Set<MetadataEntry> entries : metadata.values()) {
+			for (MetadataEntry entry : entries) {
+				fields.add(entry.getField());
+			}
+		}
+
+		/*
+		 * for each field check if the set of fields contain any they shouldn't be able to read.
+		 * this will return true if the user is allowed to read all the fields in the set.
+		 */
+		boolean allFieldsValid = fields.stream()
+				.filter(field -> {
+					//if we have a restriction on a field, compare it against the user's role on the project
+					if (restrictionMap.containsKey(field)) {
+						MetadataRestriction metadataRestriction = restrictionMap.get(field);
+						ProjectRole restrictionRole = metadataRestriction.getLevel();
+
+						//compare the restriction level to the user's role
+						if (userProjectRole.getLevel() >= restrictionRole.getLevel()) {
```

Comment:
Couldn't this be simplified to just 

```java
userProjectRole.getLevel() <= restrictionRole.getLevel()
```

Reply:
Well sure if you don't like giant unnecessary `if` blocks.  Simplified in 8ee46ca3a5